### PR TITLE
FreeBSD port: Check for Unix

### PR DESCRIPTION
**Pull request by raphaelahrens**
a lot of the libraries which are required to build Tempest on Linux are not Linux specific and are also available on other Unix like OSs (FreeBSD, OpenBSD, ...)

This commit changes the checks from Linux to Unix.

This commit has not been tested on the other platforms.

FreeBSD by default comes whit OSS and not ALSA.
The build flags were copied from the OpenAL-soft port.

### CODE REVIEW

File: Engine/CMakeLists.txt [hunk 1]
```diff
@@ -84,6 +84,16 @@ add_definitions(-DSTB_TRUETYPE_IMPLEMENTATION)
 add_definitions(-DSTB_IMAGE_WRITE_IMPLEMENTATION)
 
 ### OpenAl
+if(${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
```
Comment (Try) [inline Engine/CMakeLists.txt:87]: Is this related to openal 1.22, or it's own thing?

Comment (raphaelahrens) [reply Engine/CMakeLists.txt:87]: I copied these from the port of openal-soft for FreeBSD.
https://cgit.freebsd.org/ports/tree/audio/openal-soft/Makefile#n18

Mainly to not use ALSA by default and instead go with OSS.

**Comment by raphaelahrens** [comment]
mhhh ok after some back and forth I managed so that the build only fails with mingw.
But here I  reached my limit. Maybe mingw is considered a unix?
I must say this experience is quite frustrating and the error unhelpful.

**Comment by Try** [comment]
I think, this because openal 1.23 - error log complain about `[[unlikely]]`, what is C++20 feature

**Comment by Try** [comment]
Merged, thanks!

**Pull request merged**